vignettes/vignette_best_practices.Rmd

```diff
@@ -0,0 +1,151 @@
+
+This guidance ensures consistency across the vignettes in the admiral package in terms of content, structure and code execution. 
+As a general rule, the vignette workflow defined in [r-pkgs.org](https://r-pkgs.org/vignettes.html) is followed.
+
+
+# Metadata
+
+Each vignette in admiral should start with the following metadata.
+
+```
+---
+title: "<Title>"
+output:  rmarkdown::html_vignette:
+vignette: >
+  %\VignetteIndexEntry{<Title>}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+```
+
+The `<Title>` of the vignette should be meaningful. 
+
+# Markdown
+
+## Default options
+
+    `r ''````{r setup, include=FALSE}
+    knitr::opts_chunk$set(
+      collapse = TRUE,
+      comment = "#>"
+      )
+    ```
+
+## Required Packages
+
+    `r ''````{r, warning=FALSE, message=FALSE}
+    library(admiral)
+    #<all packages used in the vignette>
+    ```
+The `warning = FALSE` and `message = FALSE` options are there to prevent the usual messages:
+
+<span style="color: red;font-family: textmate; font-size = 12pt;" >Attaching package: 'xxxx'</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">The following objects are masked from 'package:yyyyy':</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">  fun1, fun2</span>
+
+
+
+## Format Sections
+
+### Table of contents
+
+Headings must start from Heading 1:
+
+<span style="color: red;" > **Note: some vignettes starts the heading at H2** </span>
+
+```
+# Heading 1
+## Heading 2
+
+```
+
+The first section (# Introduction) gives a brief introduction.
+
+For vignettes describing an ADaM workflow, the second section will summarize the programming workflow.
+The first sub-section within this workflow will describe the data read to demonstrate the use of the 
+admiral functions, while the last sub-section should link to a template script. 
+
+```
+## Programming Workflow
+
+* [Read in Data](#readdata)
+* [Derive/Impute End and Start Analysis Date/time and Relative Day](#datetime)
+* ...
+* [Assign `ASEQ`](#aseq)
+* [Example Script](#example)
+
+```
+
+### Conventions
+
++ Each sub-section within the programming workflow should be tagged (e.g. [Step1] (#step)), so that 
+the user can go on the relevant section from the programming workflow (in addition to the Table of
+contents). Don’t use a tag with a number but use a meaningful name (e.g. do not use `(#link1)`,
+but use `(#this_action)`)
+
++ Any variable name, dataset name, function, argument name must be quoted with backticks: e.g. "`ADSL`, `dm`, `ADT`".
+
++ Functions must also end with `()`.
+
++ All the codes must be executed and the output result displayed once the code is executed. Use:
+ 
+
+      `r ''````{r}
+      #<code>
+      ```
+
+
++ Any output created must clearly show what the function has derived. 
+It should at least show the variables/records used as input to the function and the derived variables/records.
+
+<span style="color: red;" > **Use of dataTable to show the result???** </span>
```

Comment:
I just found out that the `DT` package provides some nice extension to control the displayed columns:
```
  extensions = c('Buttons', 'ColReorder'),
  options = list(
    searchHighlight = TRUE,
    searching = TRUE,
    pageLength = 5,
    lengthMenu = c(5, 10, 15, 20, 50, 100),
    dom = 'Bfrtip',
    buttons = I('colvis'),
    colReorder = TRUE
  )
```
These enable the user to select which columns are displayed and to reorder the columns.

Reply:
Updated as suggested. I have also created a wrapper function to output the dataset in the vignette with the updated suggested code: this is currently called `dataset_vignette` but this can be updated. Please have a look.
Also, i am not sure why this new function is not  found when i build the the vignette :
```
Quitting from lines 188-193 (vignette_best_practices.Rmd) 
Error: processing vignette 'vignette_best_practices.Rmd' failed with diagnostics:
could not find function "dataset_vignette"
```
 I have tried to set `devtools::load_all()` but it still fails with the CMD checks :( Any idea/suggestion?

Reply:
I think the `@export` tag is missing in the header of the `dataset_vignette()` function.

Reply:
thanks!